wit/types.wit

```diff
@@ -16,29 +25,50 @@ interface messaging-types {
     /// The interface doesn't highlight this difference in the type itself as that's uniquely a consumer issue.
     type channel = string;
   
-    /// Configuration includes a required list of channels the guest is subscribing to, and an optional list of extensions key-value pairs 
-    /// (e.g., partitions/offsets to read from in Kafka/EventHubs, QoS etc.).
+    /// Configuration includes a required list of channels the guest is subscribing to, and an
+    /// optional list of extensions key-value pairs (e.g., partitions/offsets to read from in 
+    /// Kafka/EventHubs, QoS etc.).
     record guest-configuration {
         channels: list<channel>,
         extensions: option<list<tuple<string, string>>>
     }
   
-    /// Format specification for messages 
-    ///  - more info: https://github.com/clemensv/spec/blob/registry-extensions/registry/spec.md#message-formats
-    ///  - message metadata can further decorate w/ things like format version, and so on.
-    enum format-spec {
-        cloudevents,
-        http,
-        amqp,
-        mqtt,
-        kafka,
-        raw
-    }
-  
-    /// A message with a binary payload, a format specification, and decorative metadata.
-    record message {
-        data: list<u8>,
-        format: format-spec,
-        metadata: option<list<tuple<string, string>>>
+    /// A message with a binary payload and additional information
+    resource message {
```

Comment:
Many of the messaging implementations have the idea of a Time to Live (ttl) on a message. For example, upon receiving a message for processing, the ttl would be specified on the message metadata and the broker will lock the message in the queue so no other consumer can process the message. To ensure the message is not held forever, the broker will time bound the activity with a ttl. If the message is not ack'd in that time period, the broker will release the lock on the message and allow another consumer to receive the message.

How would a consumer discover the ttl? Would it be in the metadata? Is it up to each implementor to define how metadata is represented / interpreted?

Reply:
Normally I've seen the ttl only when you're doing the more complex patterns with things like work queues and anything else that uses delivery guarantees. As far as I understood, we're not covering those patterns right now. I think anyone can put whatever they want on metadata, but having the abandon/complete message is enough to provide an acknowledgement. Right now this has to be generic enough to support basic eventing which requires no acknowledgement or guarantees, and anything can subscribe to it. I think one of the first things I'd want to add once we get implementations for this in place would be to create a new interface (or messaging-extension package) that has extension methods for the `message` allowing you to fetch things like TTL and other details for guaranteed delivery

For request-reply, we do support a timeout for an operation so that is supported as well

Reply:
sgtm